backend/src/main/java/co/inventorsoft/scripty/service/UserServiceImpl.java

```diff
@@ -99,4 +105,17 @@ private VerificationToken generateNewVerificationToken(final User user) {
                .map(tokenRepository::save)
                .orElseThrow(()-> new ApplicationException("Token not found", HttpStatus.OK));
     }
+
+    private void createResetPasswordToken(final User user, final String passwordToken){
+        final PasswordToken resetPasswordToken = new PasswordToken(passwordToken, user);
+        passwordTokenRepository.save(resetPasswordToken);
+    }
+
+    public void sendResetPasswordToken(final EmailDto emailDto){
+        final User user = userRepository.findByEmail(emailDto.getEmail())
+                .orElseThrow(()-> new ApplicationException("Email is not found", HttpStatus.OK));
```

Comment:
Similar to verification process, we shouldn't fail for security reasons. So, create and send token only if user exists, but do not throw exception otherwise

Reply:
I need to handle the situation when the user is absent in the db so i throw an exception with status 200 and the process isn't  failing. Similar approach i used with verification process.